### DISCUSSION TITLE
Item's documentation from the formatting.format function in nvim-cmp

### DISCUSSION THREAD

**Discussion by kokou2kpadenou** [hrsh7th/nvim-cmp#1451] [Q&A]
Can an item's documentation be accessed from the formatting.format function in nvim-cmp? 
Specifically, I would like to retrieve the documentation for a completion item and include it in the formatted output.

thanks.

**Comment by proatgram** [comment]
Have you figured this out? I am trying to find out a way to format the documentation window to convert Doxygen commenting styles to Markdown to make them look better.

**Comment by kokou2kpadenou** [comment]
Hi @proatgram,

I'm sorry for not getting back to you sooner.

I have found a direct way to access an item's documentation from the formatting.format function in nvim-cmp.

In my case, on nvim-cmp's formatting option, I put the following code:

``` lua
formatting = {
  format = function(entry, vim_item)
    -- Access item documentation
    local doc = entry.completion_item.documentation

    --[[ The rest of my code go here.
    In my case, I formated the kind field based on the doc. ]]

  end,
}

```

I don't know if it will help, but you can check this part of the nvim-cmp's documentation for more details.
[https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt#L865-L881](https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt#L865-L881)
